lib/socket.ts

```diff
@@ -164,34 +159,30 @@ export class Socket extends EventEmitter {
   }
 
   /**
-   * Pings client every `this.pingInterval` and expects response
-   * within `this.pingTimeout` or closes connection.
+   * Pings client every `this.pingInterval`
    *
    * @api private
    */
   private schedulePing() {
-    clearTimeout(this.pingIntervalTimer);
     this.pingIntervalTimer = setTimeout(() => {
       debug(
         "writing ping packet - expecting pong within %sms",
         this.server.opts.pingTimeout
       );
       this.sendPacket("ping");
-      this.resetPingTimeout(this.server.opts.pingTimeout);
```

Comment:
The behavior is changed here, isn't it? The client would now have up to `pingTimeout + pingInterval` ms to respond to the ping, instead of `pingTimeout` ms in the current impl.